### PR TITLE
Update to blazor 3.0.0-preview6.19307.2

### LINKED ISSUE #29: Not working with preview6
2>    Unhandled Exception: Mono.Linker.MarkException: Error processing method: 'System.Void BlazorSignalR.Internal.BlazorServerSentEventsTransport/<ProcessEventStream>d__25::MoveNext()' in assembly: 'BlazorSignalR.dll' ---> Mono.Cecil.ResolutionException: Failed to resolve System.Void Microsoft.JSInterop.DotNetObjectRef::.ctor(System.Object)
2>       at Mono.Linker.Steps.MarkStep.HandleUnresolvedMethod(MethodReference reference)
2>       at Mono.Linker.Steps.MarkStep.MarkMethod(MethodReference reference)
2>       at Mono.Linker.Steps.MarkStep.MarkInstruction(Instruction instruction)
2>       at Mono.Linker.Steps.MarkStep.MarkMethodBody(MethodBody body)
2>       at Mono.Linker.Steps.MarkStep.ProcessMethod(MethodDefinition method)
2>       at Mono.Linker.Steps.MarkStep.ProcessQueue()
2>       --- End of inner exception stack trace ---
2>       at Mono.Linker.Steps.MarkStep.ProcessQueue()
2>       at Mono.Linker.Steps.MarkStep.ProcessPrimaryQueue()
2>       at Mono.Linker.Steps.MarkStep.Process()
2>       at Mono.Linker.Steps.MarkStep.Process(LinkContext context)
2>       at Mono.Linker.Pipeline.ProcessStep(LinkContext context, IStep step)
2>       at Mono.Linker.Pipeline.Process(LinkContext context)
2>       at Mono.Linker.Driver.Run(ILogger customLogger)
2>       at Mono.Linker.Driver.Execute(String[] args, ILogger customLogger)
2>       at Mono.Linker.Driver.Main(String[] args)

### PR DESCRIPTION
**Pull request by LunicLynx**
Fixes #29 

Please review

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -50,7 +52,7 @@ JS implemented means that the network requests are proxied to and from Javascrip
 ## Versions
 | Blazor         | BlazorSignalR |
 | --------------:| -------------:|
-| 3.0.0-preview4 |     0.6.x     |
+| 3.0.0-preview6 |     0.6.x     |
```
Comment (csnewman) [inline README.md]: 3.0.0-preview6 will be version 0.7.x

File: src/BlazorSignalR.JS/BlazorSignalR.JS.csproj [hunk 2]
```diff
@@ -11,12 +11,12 @@
 
     <!-- VS's FastUpToDateCheck doesn't consider .ts file changes, so it's necessary to disable it to get incremental builds to work correctly (albeit not as fast as if FastUpToDateCheck did work for them) -->
     <DisableFastUpToDateCheck>true</DisableFastUpToDateCheck>
-    <Version>0.6.0-blazor-3.0.0-preview4-19216-03</Version>
+    <Version>0.6.0-blazor-3.0.0-preview6.19307.2</Version>
```
Comment (csnewman) [inline src/BlazorSignalR.JS/BlazorSignalR.JS.csproj]: Change to version 0.7.0

File: src/BlazorSignalR/BlazorSignalR.csproj [hunk 3]
```diff
@@ -8,16 +8,15 @@
   <PropertyGroup>
     <OutputType>Library</OutputType>
     <TargetsForTfmSpecificBuildOutput>$(TargetsForTfmSpecificBuildOutput);IncludeP2POutput</TargetsForTfmSpecificBuildOutput>
-    <Version>0.6.0-blazor-3.0.0-preview4-19216-03</Version>
+    <Version>0.6.0-blazor-3.0.0-preview6.19307.2</Version>
```
Comment (csnewman) [inline src/BlazorSignalR/BlazorSignalR.csproj]: Version change to 0.7.0

File: src/BlazorSignalR/Internal/BlazorHttpConnectionFactory.cs [hunk 4]
```diff
@@ -42,9 +42,9 @@ public BlazorHttpConnectionFactory(IOptions<BlazorHttpConnectionOptions> options
             }
         }
 
-        public Task DisposeAsync(ConnectionContext connection)
+        public async Task DisposeAsync(ConnectionContext connection)
```
Comment (csnewman) [inline src/BlazorSignalR/Internal/BlazorHttpConnectionFactory.cs:45]: There is no need for this change

**Review by csnewman — CHANGES_REQUESTED** [review]
Some minor changes, but generally great

File: src/BlazorSignalR/Internal/BlazorHttpConnectionFactory.cs (hunk 4 above)
Comment (LunicLynx) [reply src/BlazorSignalR/Internal/BlazorHttpConnectionFactory.cs:45]: `((BlazorHttpConnection)connection).DisposeAsync()` returns ValuesTask, this was the easiest conversion i knew. Is there another way?

File: src/BlazorSignalR/BlazorSignalR.csproj [hunk 5]
```diff
@@ -8,20 +8,21 @@
...
+    <PackageReference Include="Microsoft.AspNetCore.Blazor" Version="3.0.0-preview6.19307.2" />
+    <PackageReference Include="Microsoft.AspNetCore.Components.Browser" Version="3.0.0-preview6.19307.2" />
+    <PackageReference Include="Microsoft.AspNetCore.SignalR.Client" Version="3.0.0-preview6.19307.2" />
+    <PackageReference Include="Microsoft.Extensions.Logging" Version="3.0.0-preview6.19304.6" />
   </ItemGroup>
 
   <ItemGroup>
-    <ProjectReference Include="..\BlazorSignalR.JS\BlazorSignalR.JS.csproj" />
+    <ProjectReference Include="..\BlazorSignalR.JS\BlazorSignalR.JS.csproj">
+      <PrivateAssets>All</PrivateAssets>
```
Comment (LunicLynx) [inline src/BlazorSignalR/BlazorSignalR.csproj]: I'm not sure about this. Without this change the BlazorSignalR package got a dependency on the BlazorSignalR.JS package, which does not exist. But there might be another way... 🤔

Comment (LunicLynx) [inline src/BlazorSignalR/BlazorSignalR.csproj:24]: I'm not sure about this. Without this change the BlazorSignalR package got a dependency on the BlazorSignalR.JS package, which does not exist. But there might be another way... 🤔

**Comment by kiyote** [comment]
Looking forward to this landing.

File: src/BlazorSignalR/BlazorSignalR.csproj (hunk 5 above)
Comment (csnewman) [reply src/BlazorSignalR/BlazorSignalR.csproj:24]: That's fine, I've been manually correcting the issue however this is probably the right solution

**Comment by csnewman** [comment]
Published on NuGet.

**Pull request merged**